skhep/math/vectors.py

```diff
@@ -306,3 +308,236 @@ def __repr__(self):
     def __str__(self):
         """Simple class representation."""
         return str(tuple(self.__values))
+
+
+#-----------------------------------------------------------------------------
+# Lorentz vector class
+#-----------------------------------------------------------------------------
+class LorentzVector(object):
+    """
+    Class representing a Lorentz vector,
+    either a 4-dimensional Minkowski space-time vector or a 4-momentum vector.
+    The 4-vector components can be seen as (x,y,z,t) or (px,py,pz,E).
+
+    Constructors:
+        __init__(x=0., y=0., z=0., t=0.)
+        from4vector(avector)
+        from3vector(vector3D, t)
+    """
+    def __init__(self, x=0., y=0., z=0., t=0.):
+        """Default constructor.
+
+        :Example:
+
+        >>> v1 = LorentzVector()
+        >>> v1
+        """
+        self.__values = [x,y,z,t]
```

Comment:
See the comment "For reference here he had already agreed to improve the storing of the vector(s) with NumPy arrays rather than simple lists." and others in the description of this PR. Feel free to update to a NumPy array arlready now.